### DISCUSSION TITLE
Admin API Send Verify Email doesnt send emailVerification Mail, but executeActions Mail

### DISCUSSION THREAD

**Discussion by buti1021** [keycloak/keycloak#10955] [Feedback]
The /send-verify-mail endpoint of the Admin API (https://www.keycloak.org/docs-api/17.0/rest-api/index.html#_sendverifyemail)
is not sending the text configured in the emailVerification section of the message_**.properties, but the texts of the executeActions sections. 
While I can see that the wording of those texts might make more sense in some use cases, I think this is clearly not fitting the documentation and the executeActions mails can already be sent by the /execute-actions-email endpoint https://www.keycloak.org/docs-api/17.0/rest-api/index.html#_executeactionsemail

**Comment by rcapp** [comment]
UPDATE (as of 14/02/2023): 

For future reference in case someone else gets here: 

https://issues.redhat.com/browse/KEYCLOAK-11686
https://issues.redhat.com/browse/KEYCLOAK-5034



These 2 links refer to this "issue". Because the expected behaviour for someone new to Keycloak is to receive/send an email when the `send-verify-email` is called, people may report/comment this is from time to time.  However, as we can see on the above links, this won't be "fixed" - it is not a bug or unexpected.   

==========

Hi! 

Almost a year after this, I hit the same "problem". Plus, I can't find where I activate the code validation flow. I was looking into the e-mail templates to see how to build one and found this:

https://github.com/keycloak/keycloak/blob/main/themes/src/main/resources/theme/base/email/html/email-verification-with-code.ftl


But there's no mention on other parts of the repository to this template or the variable/parameter `emailVerificationBodyCodeHtml`. 

Thanks in advance!

**Comment by rcapp** [comment]
Hi, all!

In one of the above reports, they ask to clarify/explain why this would be an expected/desired feature. As of today, here where we are implementing Keycloak, we need different registration pages because they have small differences in the data they want/need to registre during this phase. In fact, we are implementing KC to allow us to have a SSO because the current state of registration and login is "everyone doing their own thing and then, we try to make the parts cooperate somehow".

These applications will belong to the same realm and will use their `client_id` to identify themselves and their users' request.  

As mentioned, we need users to provide slight different information to register them in their correct clients. We are using/will use the REST API to achieve this and, unless there is something really different with the JS driver and the React library, I understand every library - no matter which language - make use of the this API.  Right now, we are implementing our backend using `python-keycloak`.

When we register a user using the REST API, no e-mail verification is sent. The same goes with the current library. As said, unless there's something substantially different with the JS process, I think it should be took into account the possibility of using `emailVerification` for the  `/send-verify-email/` REST API. 


Thank you in advance!

**Comment by buti1021** [comment]
Fixed with Keycloak24
